components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/ImportExportAPI.java

```diff
@@ -31,36 +34,62 @@
-            throws APIManagementException, APIImportExportException;
+            ExportFormat format, boolean preserveDocs) throws APIManagementException, APIImportExportException;
 
     /**
      * Used to export API product artifact.
      *
-     * @param apiId UUID of API.
-     * @param name name of API.
-     * @param version version of API.
-     * @param providerName provider of API.
-     * @param format Format of output documents. Can be YAML or JSON
+     * @param apiId          UUID of API.
+     * @param name           name of API.
+     * @param version        version of API.
+     * @param providerName   provider of API.
+     * @param format         Format of output documents. Can be YAML or JSON
      * @param preserveStatus Preserve API status on export
-     * @param preserveDocs Preserve documentation on Export.
+     * @param preserveDocs   Preserve documentation on Export.
      * @return APIProduct Artifact.
      * @throws APIManagementException
      * @throws APIImportExportException
      */
-    public File exportApiProduct(String apiId, String name, String version, String providerName, ExportFormat format,
-                                 boolean preserveStatus, boolean preserveDocs)
-            throws APIManagementException, APIImportExportException;
+    public File exportAPIProduct(String apiId, String name, String version, String providerName, ExportFormat format,
+            boolean preserveStatus, boolean preserveDocs) throws APIManagementException, APIImportExportException;
+
+    /**
+     * Used to import an API artifact.
+     *
+     * @param fileInputStream  Input stream from the REST request
+     *                         (This will not be null when importing dependent APIs with API Products)
+     * @param preserveProvider Decision to keep or replace the provider
+     * @param overwrite        Whether to update the API or not
+     * @throws APIImportExportException If there is an error in importing an API
+     * @@return Imported API
+     */
+    public API importAPI(InputStream fileInputStream, Boolean preserveProvider, Boolean overwrite, String[] tokenScopes)
+            throws APIManagementException;
+
+    /**
+     * Used to import an API Product artifact.
+     *
+     * @param fileInputStream     Input stream from the REST request
+     *                            (This will not be null when importing dependent APIs with API Products)
+     * @param preserveProvider    User choice to keep or replace the API Product provider
+     * @param importAPIs          Whether to import the dependent APIs or not.
+     * @param overwriteAPIProduct Whether to update the API Product or not. This is used when updating already existing API Products.
+     * @param overwriteAPIs       Whether to update the dependent APIs or not. This is used when updating already existing dependent APIs of an API Product.
+     * @@return Imported API Product
+     */
```

Comment:
Missing values in the java docs. Please add.

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/commit/eb9b7181c0f6cb5b9624602d91d28a7479bf551b